include/boost/numeric/interval/hw_rounding.hpp

```diff
@@ -44,23 +46,51 @@ namespace boost {
 namespace numeric {
 namespace interval_lib {
 
+namespace detail
+{
+  template<class T>
+  struct runtime_checker
+  {
+  protected:
+    runtime_checker()
+    {
+      static helper _;
+    }
+
+  private:
+    struct helper
+    {
+      helper()
+      {
+        save_state<rounded_arith_opp<T> > rounding;
+        T a = -(T(-1.1) * T(10.1));
+        T b = +(T(+1.1) * T(10.1));
+        assert(a != b);
+      }
+    };
+  };
+}
+
 /*
  * Three specializations of rounded_math<T>
  */
 
 template<>
 struct rounded_math<float>
   : save_state<rounded_arith_opp<float> >
+  , private detail::runtime_checker<float>
 {};
 
 template<>
 struct rounded_math<double>
   : save_state<rounded_arith_opp<double> >
+  , private detail::runtime_checker<double>
```

Comment:
Won't this increase the runtime performance significantly if these are used in hot code paths?

Reply:
It shouldn't because it is invoking only once (`static helper _`, line 59). There is still some performance penalty due to the check that singleton of the type `helper` is initialized, but I believe that it's insignificant in comparison to the cost of switch of rounding mode.